src/overlap_check/ProgressBar.cpp

```diff
@@ -0,0 +1,68 @@
+
+
+#include "ProgressBar.hpp"
+
+#include <sstream>
+#include <iostream>
+#include <iomanip>
+
+#if defined(__unix__) || defined(__unix) || (defined(__APPLE__) && defined(__MACH__))
+#include <unistd.h>
+#endif
+
+#define BAR_WIDTH 72
+
+ProgressBar::ProgressBar() {
+  // initialize bar
+  set_value(0.0);
+}
+
+ProgressBar::~ProgressBar() {
+  std::cout << "\n";
+}
+
+bool ProgressBar::is_terminal() {
+#ifdef _POSIX_VERSION
+  return isatty(STDOUT_FILENO) != 0;
+#else
+  return false;
+#endif
+}
+
+
+void ProgressBar::set_value(double val) {
+
+  if (!is_terminal()) {
```

Comment:
Couldn't this check be done in the constructor?  It won't change during the run, right?

Reply:
Obviously, you'd still need to check something for an early exit, but instead of running `isatty()` each time, you could just check a boolean member variable.